docs/content/latest/api/ysql/datatypes/type_datetime/date-time-data-types-semantics/type-interval/interval-arithmetic/interval-interval-comparison.md

```diff
@@ -0,0 +1,289 @@
+
+```output
+<  <=  = >=  >  !=
+```
+
+## Modeling the interval-interval comparison test
+
+Implement the model as _function modeled_inequality_comparison()_ thus. It depends upon these two functions, defined in the _"User-defined interval utility functions"_ section:
+
+- _[interval_mm_dd_ss (interval)](../../interval-utilities/#function-interval-mm-dd-ss-interval-returns-interval-mm-dd-ss-t)_
+
+- _[justified_seconds(interval)](../../interval-utilities/#function-justified-seconds-interval-returns-double-precision)_
+
+```plpgsql
+drop function if exists modeled_inequality_comparison(interval, interval) cascade;
+
+create function modeled_inequality_comparison(i1 in interval, i2 in interval)
+  returns text
+  language plpgsql
+as $body$
+declare
+  gt constant text not null := 'i1 > i2';
+  eq constant text not null := 'i1 = i2';
+  lt constant text not null := 'i1 < i2';
+
+  -- This implicity tests that neither "i1" nor "i2" is NULL.
+  actual_ineq   constant text             not null := case
+                                                        when i1 > i2 then gt
+                                                        when i1 = i2 then eq
+                                                        when i1 < i2 then lt
+                                                      end;
+
+  s1            constant double precision not null := justified_seconds(i1);
+  s2            constant double precision not null := justified_seconds(i2);
+  modeled_ineq  constant text             not null := case
+                                                        when s1 > s2 then gt
+                                                        when s1 = s2 then eq
+                                                        when s1 < s2 then lt
+                                                      end;
+begin
+  assert modeled_ineq = actual_ineq, 'Assert #1 failed';
+
+  -- Test an alternative model for equality.
+  if actual_ineq = eq then
+    declare
+      r1 constant interval_mm_dd_ss_t not null := interval_mm_dd_ss(justify_interval(i1));
+      r2 constant interval_mm_dd_ss_t not null := interval_mm_dd_ss(justify_interval(i2));
+    begin
+      assert r1 = r2, 'Assert #2 failed';
+    end;
+  end if;
+
+  return actual_ineq;
+end;
+$body$;
+```
+
+Notice the use of the _assert_ statement to self-test an alternative definition of the model that works only for the `=` comparison.
+
+Sanity-check the model by rewriting the code example in the [parent section](../../interval-arithmetic#the-interval-interval-overload-of-the-operator):
+
+```plpgsql
+select
+  modeled_inequality_comparison(
+    '5 days  1 hours'::interval,
+    '4 days 25 hours'::interval
+  )::text as e1,
+  modeled_inequality_comparison(
+    '5 months  1 day' ::interval,
+    '4 months 31 days'::interval
+  )::text as e2;
+```
+
+It finishes silently, showing that the _assert_ holds, and produces this result:
+
+```
```

Comment:
Make sure to give all code blocks a language. In this case, this one should be "output"
````suggestion
```output
````